MobiFlight/MidiBoard/MidiBoardManager.cs

```diff
@@ -0,0 +1,273 @@
+﻿using M;
+using MobiFlight.UI.Dialogs;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Windows.Forms;
+
+namespace MobiFlight
+{
+    public class MidiBoardManager
+    {
+        public event EventHandler Connected;
+        public event ButtonEventHandler OnButtonPressed;      
+        private readonly List<MidiBoard> MidiBoards = new List<MidiBoard>();
+        private readonly List<MidiBoard> BoardsToBeRemoved = new List<MidiBoard>();
+        private readonly Dictionary<string, MidiBoardDefinition> Definitions = new Dictionary<string, MidiBoardDefinition>();
+        private readonly Timer ProcessTimer = new System.Windows.Forms.Timer();
+        private int CheckAttachedRemovedCounter = 0;
+
+        public MidiBoardManager ()
+        {
+            LoadDefinitions();
+            ProcessTimer.Interval = 50;
+            ProcessTimer.Tick += ProcessTimer_Tick;
+        }
+
+        private List<string> CollectErrorsInInputDefinition(MidiBoardDefinition def)
+        {
+            List<string> errorMessages = new List<string>();
+            if (def.Inputs == null)
+                errorMessages.Add($"No input definitions.");
+            else
+            {
+                foreach (var item in def.Inputs)
+                {
+                    if (item.LabelIds.Length != item.MessageIds.Length)
+                        errorMessages.Add($"{item.Label}: Number of LabelIds unequal to number of MessageIds");
+                    if (!item.Label.Contains("%"))
+                        errorMessages.Add($"{item.Label}: Label does not contain % placeholder.");
+                }
+            }
+            return errorMessages;
+        }
+
+        private List<string> CollectErrorsInOutputDefinition(MidiBoardDefinition def)
+        {
+            List<string> errorMessages = new List<string>();
+            if (def.Outputs != null)
```

Comment:
I don't believe this if statement is necessary. The following foreach just won't iterate if `def.Outputs` is null.

Reply:
Hmm, it is necessary. Instead this should be `if (def.Outputs == null) return errorMessages`;

Reply:
Ok, this is a question of style and taste. I'm in the camp which likes only 1 return statement per function. In the end it is a minor thing and not worth endless discussions. Here some literature on that topic :-).

https://hackerchick.com/religious-war-48293-single-vs-multiple/
https://stschindler.medium.com/having-only-one-return-statement-is-better-and-more-safe-5cd379b7e87c
https://www.anthonysteele.co.uk/TheSingleReturnLaw.html

Reply:
I agree it's a matter of style. I read those and I don't think any of them argued against what I suggested though 🤣 Ultimately it's up to @DocMoebiuz but the current codebase does tend to prefer early returns/continues over nested ifs when it comes to validating variables before executing the core part of a function.

Reply:
Yes, I know. I justed wanted to show that there is an argument going on. In the end short methods are most important and prevent most crazy stuff. Rest is small optimizations.

Reply:
I defintely prefer early returns, it makes the code easier to read. Please adhere to the same style whenever possible.